File: Documentacao/LIA2_008_PedagogicalAdaptive_Mobile_Thinking_FIX02/apps/studentWeb/src/contracts/studyScopeContract.ts
export interface StudyScopeContract {
  contractName: 'StudyScope.v1'
  studyScopeId: string
  learningGoalId: string
  name: string
  description: string | null
  status: 'DRAFT' | 'ACTIVE' | 'COMPLETED' | 'ARCHIVED'
  createdAt: string
  updatedAt: string
}

export interface StudyScopeCreateContract {
  contractName: 'StudyScopeCreate.v1'
  name: string
  description: string | null
}

File: apps/backend/app/contracts/studyScopeContract.py
from datetime import datetime
from typing import Literal
from uuid import UUID
from pydantic import BaseModel, ConfigDict, Field, field_validator

class StudyScopeCreateContract(BaseModel):
    contractName: Literal["StudyScopeCreate.v1"] = "StudyScopeCreate.v1"
    name: str = Field(min_length=2, max_length=250)
    description: str | None = Field(default=None, max_length=1500)

    @field_validator("name", "description")
    @classmethod
    def normalizeText(cls, value):
        if value is None: return None
        normalized = " ".join(value.split())
        return normalized or None

class StudyScopeContract(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    contractName: Literal["StudyScope.v1"] = "StudyScope.v1"
    studyScopeId: UUID
    learningGoalId: UUID
    name: str
    description: str | None
    status: Literal["DRAFT","ACTIVE","COMPLETED","ARCHIVED"]
    createdAt: datetime
    updatedAt: datetime
